src/Glpi/Asset/AssetDefinition.php

```diff
@@ -1161,4 +1166,38 @@ private function validateProfileArray(mixed $profiles, bool $check_values = true
 
         return $is_valid;
     }
+
+    /**
+     * @return CustomField[]
+     */
+    public function getCustomFieldDefinitions(): array
+    {
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        static $fields = null;
```

Comment:
Please do not use a `static` variable here. Indeed, may be a pain in testing context, and all the instances of the class will share the same value, and that will probably be a problem.

As the method is not static, you can store the values ina `private array $custom_fields_definition = null;` class property and fill it `if ($this->custom_fields_definition === null)`.